src/Symfony/Component/Message/README.md

```diff
@@ -0,0 +1,250 @@
+the message bus with Symfony's FrameworkBundle, the following middlewares are configured for you:
+
+1. `LoggingMiddleware` (log the processing of your messages)
+2. `SendMessageMiddleware` (enable asynchronous processing)
+3. `HandleMessageMiddleware` (call the registered handle)
+
+```php
+use App\Message\MyMessage;
+
+$result = $this->get('message_bus')->handle(new MyMessage(/* ... */));
+```
+
+### Handlers
+
+Once dispatched to the bus, messages will be handled by a "message handler". A message handler is a PHP callable 
+(i.e. a function or an instance of a class) that will do the required processing for your message. It _might_ return a
+result.
+
+```php
+namespace App\MessageHandler;
+
+use App\Message\MyMessage;
+
+class MyMessageHandler
+{
+    public function __invoke(MyMessage $message)
+    {
+        // Message processing...
+    }
+}
+```
+
+```xml
+<service id="App\Handler\MyMessageHandler">
+    <tag name="message_handler" />
+</service>
+```
+
+### Asynchronous messages
+
+Using the Message Component is useful to decouple your application but it also very useful when you want to do some
+asychronous processing. This means that your application will produce a message to a queuing system and consume this
+message later in the background, using a _worker_.
+
+#### Adapters
+
+The communication with queuing system or 3rd parties is for delegated to libraries for now. You can use one of the 
+following adapters:
+
+- [PHP Enqueue bridge](https://github.com/sroze/enqueue-bridge) to use one of their 10+ compatible queues such as 
+  RabbitMq, Amazon SQS or Google Pub/Sub.
+
+#### Routing
+
+When doing asynchronous processing, the key is to route the message to the right sender. As the routing is
+application-specific and not message-specific, the configuration can be made within the `framework.yaml` 
+configuration file as well:
+
+```yaml
+framework:
+    message:
+        routing:
+            'My\Message\MessageAboutDoingOperationalWork': my_operations_queue_sender
+```
+
+Such configuration would only route the `MessageAboutDoingOperationalWork` message to be asynchronous, the rest of the
+messages would still be directly handled.
+
+If you want to do route all the messages to a queue by default, you can use such configuration:
+```yaml
+framework:
+    message:
+        routing:
+            'My\Message\MessageAboutDoingOperationalWork': my_operations_queue_sender
+            '*': my_default_sender
+```
+
+Note that you can also route a message to multiple senders at the same time:
+```yaml
+framework:
+    message:
+        routing:
+            'My\Message\AnImportantMessage': [my_default_sender, my_audit_semder]
+```
+
+#### Same bus received and sender
+
+To allow us to receive and send messages on the same bus and prevent a loop, the message bus is equipped with the
+`WrappedIntoReceivedMessage` received. It will wraps the received messages into `ReceivedMessage` objects and the 
```

Comment:
`WrapIntoReceivedMessage` + `It will wrap the..`

Reply:
Yep, good point 👍